api/projects/tasks.py

```diff
@@ -6,3 +8,17 @@ def write_environments_to_dynamodb(project_id: int):
     from environments.models import Environment
 
     Environment.write_environments_to_dynamodb(project_id=project_id)
+
+
+@register_task_handler()
+def migrate_project_environments_to_v2(project_id: int):
+    from environments.dynamodb.services import migrate_environments_to_v2
+    from projects.models import IdentityOverridesV2MigrationStatus, Project
+
+    with transaction.atomic():
+        project = Project.objects.select_for_update().get(id=project_id)
```

Comment:
Why `select_for_update` here? The project is used heavily in the hot paths and the documentation is kind of unclear as to whether the row is locked for reads or not.

Reply:
> Why `select_for_update` here? 

This is the best way I could think of to avoid accidentally spamming the task.

The row is not locked for reads in accordance with default [read committed](https://www.postgresql.org/docs/7.2/xact-read-committed.html) isolation level. This is easy to check e.g. via two Django shells. Unless some of our hot paths involve writing to a project row, we're good.